model/src/main/java/com/redhat/ipaas/model/connection/Connection.java

```diff
@@ -49,7 +50,7 @@ default String getKind() {
 
     Optional<String> getConnectorId();
 
-    String getConfiguredProperties();
+    Map<String, String> getConfiguredProperties();
```

Comment:
Sounds good to me!

We can keep the form config as-is though with type support as that'll provide a better experience in the UI.